client/dashboard/sites/monitoring-request-methods-card/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { siteMetricsQuery } from '@automattic/api-queries';
+import { DataPointPercentage, PieChart, Legend } from '@automattic/charts';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import MonitoringCard from '../monitoring-card';
+import type { LegendData, PeriodData, TimeRange } from '../monitoring/types';
+import type { Site, SiteHostingMetrics } from '@automattic/api-core';
+
+function convertTimeRangeToUnix( timeRange: number ): TimeRange {
+	const start = Math.floor( new Date().getTime() / 1000 ) - timeRange * 3600;
+	const end = Math.floor( new Date().getTime() / 1000 );
+
+	return { start, end };
+}
+
+type SiteRequestMethodsData = {
+	data: DataPointPercentage[];
+	isLoading: boolean;
+};
+
+const chartColors = [ '#3858E9', '#5BA300', '#F57600', '#B51963' ];
+
+function useSiteRequestMethodsData( siteId: number, timeRange: number ): SiteRequestMethodsData {
+	const { start, end } = useMemo( () => convertTimeRangeToUnix( timeRange ), [ timeRange ] );
+
+	const { data: requestMethodsData, isPending: isLoading } = useQuery(
+		siteMetricsQuery( siteId, { start, end, metric: 'requests_persec', dimension: 'http_verb' } )
```

Comment:
I think we can add `select` function here to format the data 🙂

```js
useQuery(
  ...siteMetricsQuery(),
  select: ( data ) => {
    // format the data
  }
)
```

Reply:
Done: https://github.com/Automattic/wp-calypso/pull/105863/commits/84fb49872180bfdb4fdcd2336d4511dbf7aca2c3